src/component/RecipeList.js

```diff
@@ -0,0 +1,39 @@
+import { useState } from "react";
+import RecipeListItem from "./RecipeListItem";
+import Pagination from "./Pagination";
+
+function RecipeList({ recipes, handleChooseRecipe }) {
+  const [currentPage, setCurrentPage] = useState(1);
+  //   const [recipeList, setRecipeList] = useState([
+  //     ...recipes.slice(startIndex, endIndex),
+  //   ]);
+  const [recordsPerPage] = useState(10);
+
+  const endIndex = currentPage * recordsPerPage;
+  const startIndex = endIndex - recordsPerPage;
+  const nPages = Math.ceil(recipes.length / recordsPerPage);
+  const recipeList = recipes.slice(startIndex, endIndex);
+
+  const renderedRecipes = recipeList.map((recipe) => (
+    <RecipeListItem
+      key={recipe.id}
+      recipe={recipe}
+      handleChooseRecipe={handleChooseRecipe}
+    />
+  ));
+
+  return (
+    <div>
+      <div>{renderedRecipes}</div>
+      {renderedRecipes.length !== 0 && (
```

Comment:
Why do you need this check? Also, this still shows the pagination button if 0 < `recipeList.length` > 10, so it isn't correct here 🤔

Reply:
Because an empty renderedRecipes array is truthy in this conditional rendering which allows rendering the pagination even when I haven't searched for any recipe.